code/datums/components/buckle_mob_effect.dm

```diff
@@ -0,0 +1,64 @@
+
+/**
+ * # Buckle mob effect component
+ *
+ * A mob with this component will continously apply a callback while its buckled to a mob
+ * Intended for slimes draining your life force, leeches drinking your blood, snakes constricting you
+ *
+ */
```

Comment:
I will say i'm not super sold on this being a component 

I think this could easily be a status effect applied via `/mob/basic/slime/post_buckle_mob(mob/living/buckled_mob)` to the buckled mob

Then the status effect itself could handle clearing itself when the slime unbuckles (either manually in `post_unbuckle_mob` or via signal, the latter's maybe more clean) 

```dm
/datum/status_effect/slime_leech
    ...
    var/mob/living/slime

/datum/status_effect/slime_leech/on_creation(mob/living/new_owner, mob/living/slime)
    src.slime = slime
    return ..()

/datum/status_effect/slime_leech/on_apply()
    if(isnull(slime))
        return FALSE
    
    // Register a bunch of signals for checking the slime unbuckling / being deleted / etc
    ...

/datum/status_effect/slime_leech/tick()
    // Do the effects
    ...
```
```dm
/mob/basic/slime/post_buckle_mob(mob/living/buckled_mob)
    . = ..()
    buckled_mob.apply_status_effect(/datum/status_effect/slime_leech, src)
````

Reply:
Ah, I wanted to make it a generic component, so it could be applied to say, leeches draining blood and anacondas constricting folks, allowing us to keep the mechanics contained in the buckling mobs themselves, while the component would keep the mininal code needed.

Reply:
I have converted it to a status effect as discussed, with some small changes from the plan.
I have placed the status effect application in `start_feeding()`, after buckling returns true, because the suggested `post_buckle_mob` would not have had worked, as it is the other way around, its for when a mob is buckled to a slime.
I also have moved the effect that covers the target in acidic slime into the status effect, applied by the proc that is called when the slime suggested signals are triggered, because I realized, there might be sources that could force unbuckle the slime (who are normally only force buckleable). 
I might have to adjust the tick inverval of the effect, or the amount of nutrition gained, because the slimes gained only the half the growth they got when this was done by a component.

Reply:
I increased the nutrition gained, now its more in line with the previous implementation.